### DISCUSSION TITLE
SQL Trigger Azure function connection string when running it from Visual Studio

### DISCUSSION THREAD

**Discussion by basigu** [Azure/azure-functions-sql-extension#958] [Q&A] [answered]
I am running Sql Trigger Azure function from visual studio and it is failing with sql connection string.  On the sql database, sql login has been disabled.  
I have tried the following connection string:
Server=tcp:servername,1433; Authentication=Active Directory Managed Identity; Database=dbname;
**Error**: Exception: Microsoft.Data.SqlClient.SqlException (0x80131904): ManagedIdentityCredential authentication unavailable. Multiple attempts failed to obtain a token from the managed identity endpoint.

When i tried the following connection string:
Encrypt=True;Connection Timeout=30;Server=tcp:servername,1433;Database=dbname;
**Error**: Exception: Microsoft.Data.SqlClient.SqlException (0x80131904): Login failed for user ''.

What's the best right sql connection when I run sql trigger azure function from visual studio?

**Comment by lucyzhang929** [comment]
Hi @basigu, here is an article with steps to use Managed Identity with SQL Bindings: https://learn.microsoft.com/en-us/azure/azure-functions/functions-identity-access-azure-sql-with-managed-identity.

**Reply by basigu** [reply]
I have already looked into this documentation.  The suggested connection string Server=demo.database.windows.net; Authentication=Active Directory Managed Identity; Database=testdb does not work when I run azure function from visual studio.

**Reply by lucyzhang929** [reply, marked as answer]
Can you try "Server=demo.database.windows.net; Authentication=Active Directory Default; Database=testdb"?

**Reply by basigu** [reply]
@lucyzhang929 It worked.  Thank you!
